### DISCUSSION TITLE
Provision>KnowledgePlatform>Neo4j Destination directory conf does not exist

### DISCUSSION THREAD

**Discussion by daveanvit** [Sunbird-Ed/Community#141] [Installation] [answered]
Jenkins: Ubuntu 18.04 LTS, 4vCPU, 16GB RAM, 250 GB HDD
Github Release Tag: release-4.6.0_RC4
*Destination directory /home/learning/neo4j-learning/lp_neo4j_artifacts/conf does not exist*
Jenkins Logs:

`Started by user [admin](http://34.138.204.52:8080/user/admin)
Replayed [#11](http://34.138.204.52:8080/job/Provision/job/dev/job/KnowledgePlatform/job/Neo4j/11/)
Loading library deploy-conf@heads/shared-lib
Attempting to resolve heads/shared-lib from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.37.3'
 > git ls-remote -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j@libs/deploy-conf/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.37.3'
 > git fetch --no-tags --force --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* refs/heads/shared-lib # timeout=10
Checking out Revision 778da7ce7f97efd003eed98071921a837d209343 (778da7ce7f97efd003eed98071921a837d209343)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
Commit message: "fix: move variables to be accessible across both try blocks"
 > git rev-list --no-walk 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](http://34.138.204.52:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] {
[Pipeline] stage
[Pipeline] { (checkout public repo)
[Pipeline] echo
Found .git folder. Clearing it..
[Pipeline] sh
+ git clean -fxd
Removing ansible/artifacts/lp_neo4j_artifacts.zip:release-4.1.0_RC2_c939bcc_21
Removing private@tmp/
[Pipeline] checkout
The recommended git tool is: git
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-learning-platform.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-learning-platform.git
 > git --version # timeout=10
 > git --version # 'git version 2.37.3'
 > git fetch --tags --force --progress --depth=1 -- https://github.com/project-sunbird/sunbird-learning-platform.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.6.0_RC2^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC2^{commit} # timeout=10
Checking out Revision d708cf824572355205448d7695e9885b033c769a (release-4.6.0_RC2)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f d708cf824572355205448d7695e9885b033c769a # timeout=10
Commit message: "Issue #SB-28039 merge: Merge pull request #1657 from krgauraw/release-4.6.0"
 > git rev-list --no-walk d708cf824572355205448d7695e9885b033c769a # timeout=10
[Pipeline] }
[Pipeline] // stage
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (deploy)
[Pipeline] sh
+ pwd
[Pipeline] sh
+ echo Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] sh
+ echo Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] sh
+ echo Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] echo
{currentWs=/var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j, env=dev, module=KnowledgePlatform, jobName=Neo4j, ansiblePlaybook=/var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/lp_learning_neo4j_provision.yml, ansibleExtraArgs=--vault-password-file /var/lib/jenkins/secrets/vault-pass}
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (ansible-run)
[Pipeline] echo
{currentWs=/var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j, env=dev, module=KnowledgePlatform, jobName=Neo4j, ansiblePlaybook=/var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/lp_learning_neo4j_provision.yml, ansibleExtraArgs=--vault-password-file /var/lib/jenkins/secrets/vault-pass}
[Pipeline] dir
Running in /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/private
[Pipeline] {
[Pipeline] echo
Info: Branch override is enabled
[Pipeline] checkout
The recommended git tool is: NONE
using credential github-cred
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/private/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/daveanvit/searce-diksha-sunbird.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/daveanvit/searce-diksha-sunbird.git
 > git --version # timeout=10
 > git --version # 'git version 2.37.3'
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --force --progress --depth=1 -- https://github.com/daveanvit/searce-diksha-sunbird.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/main^{commit} # timeout=10
Checking out Revision f0d7479687eece6bba4d4f0a7beb1af458f2ba50 (origin/main)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f f0d7479687eece6bba4d4f0a7beb1af458f2ba50 # timeout=10
Commit message: "Update secrets.yml"
 > git rev-list --no-walk f0d7479687eece6bba4d4f0a7beb1af458f2ba50 # timeout=10
[Pipeline] }
[Pipeline] // dir
[Pipeline] sh
+ rsync -Lkr /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/private/ansible/inventory/dev/KnowledgePlatform/common.yml /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/private/ansible/inventory/dev/KnowledgePlatform/hosts /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/private/ansible/inventory/dev/KnowledgePlatform/secrets.yml /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/inventory/env/
+ [ -f /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/inventory/env/kubernetes.yaml ]
+ ansible-playbook -i /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/inventory/env/hosts /var/lib/jenkins/workspace/Provision/dev/KnowledgePlatform/Neo4j/ansible/lp_learning_neo4j_provision.yml --vault-password-file /var/lib/jenkins/secrets/vault-pass
[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details

PLAY [local] *******************************************************************

TASK [Gathering Facts] *********************************************************
/usr/lib/python3/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
ok: [localhost]

TASK [artifacts-download-azure : Ensure azure blob storage container exists] ***
changed: [localhost]

TASK [artifacts-download-azure : Download from azure blob storage] *************
changed: [localhost]

PLAY [local] *******************************************************************

TASK [Download neo4j community] ************************************************
skipping: [localhost]
[WARNING]: Found variable using reserved name: port

PLAY [learning-neo4j-cluster] **************************************************

TASK [Gathering Facts] *********************************************************
ok: [192.170.0.8]

TASK [Create learner group] ****************************************************
ok: [192.170.0.8]

TASK [Create learner user] *****************************************************
ok: [192.170.0.8]

TASK [Registering server_id] ***************************************************
ok: [192.170.0.8]

TASK [jdk-1.8.0_121 : Uncompressing and copying to system path] ****************
ok: [192.170.0.8]

TASK [jdk-1.8.0_121 : Updating java jdk to system path] ************************
skipping: [192.170.0.8]

TASK [jdk-1.8.0_121 : Checking java version] ***********************************
changed: [192.170.0.8]

TASK [jdk-1.8.0_121 : Java version] ********************************************
ok: [192.170.0.8] => {
    "java.stderr": "openjdk version \"1.8.0_342\"\nOpenJDK Runtime Environment (build 1.8.0_342-8u342-b07-0ubuntu1~18.04-b07)\nOpenJDK 64-Bit Server VM (build 25.342-b07, mixed mode)"
}

TASK [jdk-1.8.0_121 : Remove post install script] ******************************
ok: [192.170.0.8]

TASK [jdk11 : Uncompressing and copying to system path] ************************
ok: [192.170.0.8]

TASK [neo4j-cluster : install] *************************************************
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via 
squash_actions is deprecated. Instead of using a loop to supply multiple items 
and specifying `name: "{{item}}"`, please use `name: ['unzip']` and remove the 
loop. This feature will be removed in version 2.11. Deprecation warnings can be
 disabled by setting deprecation_warnings=False in ansible.cfg.
ok: [192.170.0.8] => (item=['unzip'])

TASK [neo4j-cluster : create folder data for learning] *************************
ok: [192.170.0.8] => (item=/data)
ok: [192.170.0.8] => (item=/data/logs)

TASK [neo4j-cluster : Create Paths] ********************************************
ok: [192.170.0.8]

TASK [neo4j-cluster : debug] ***************************************************
ok: [192.170.0.8] => {}

MSG:

System neo4j-learning

TASK [neo4j-cluster : copy the neo4j cluster zip file] *************************
ok: [192.170.0.8]

TASK [neo4j-cluster : Extract archive lp_neo4j_artifacts.zip:release-4.1.0_RC2_c939bcc_21] ***
skipping: [192.170.0.8]

TASK [neo4j-cluster : Change Permissions neo4j-enterprise-3.0.4] ***************
ok: [192.170.0.8]

TASK [neo4j-cluster : Copy the template neo4j.conf file] ***********************
fatal: [192.170.0.8]: FAILED! => {
    "changed": false,
    "checksum": "12ecadee79b373dd07430989c4e152e1a98fe7b4"
}

MSG:

Destination directory /home/learning/neo4j-learning/lp_neo4j_artifacts/conf does not exist

PLAY RECAP *********************************************************************
192.170.0.8                : ok=15   changed=1    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0   
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ rev
+ git name-rev --name-only HEAD
+ cut -d / -f1
+ rev
[Pipeline] sh
+ echo Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Provision/dev/KnowledgePlatform/Neo4j
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE`

**Comment by beepdot** [comment, marked as answer]
Hello. I believe something to do with this variable - https://github.com/project-sunbird/sunbird-devops/blob/master/private_repo/ansible/inventory/dev/KnowledgePlatform/common.yml#L23
Can you please check on the VM? What are the directories and its names under ` /home/learning/` related to neo4j?
Usually if you use the defaults, the directory will look like this - ` /home/learning/neo4j-learning/neo4j-community-3.3.9`

But in this case, its looking for `/home/learning/neo4j-learning/lp_neo4j_artifacts`

**Reply by daveanvit** [reply]
@keshavprasadms I have delete deleted folder from VM and re-ran it with defaults it worked for me.

Marking this answered thanks @keshavprasadms
